ocp_diagnosis.sh

```diff
@@ -70,51 +66,66 @@ prometheus_pod=$(oc get pods -n $prometheus_namespace | grep -w "Running" | awk
 # get the timestamp
 ts=$(date +"%Y%m%d-%H%M%S")
 
-# set the output_dir to pbench results dir
-if [[ "$STORAGE_MODE" == "pbench" ]]; then
-	echo "Detected sotrage mode as $STORAGE_MODE"
-	echo "Assuming that the ocp diagnosis tool is run using pbench-user-benchmark"
-	echo "Fetching the latest pbench results dir"
-	result_dir="/var/lib/pbench-agent/$(ls -t /var/lib/pbench-agent/ | grep "pbench-user" | head -1)"/1/reference-result
-        echo "Copying the collected data to $result_dir"
-        export OUTPUT_DIR="/var/lib/pbench-agent/$(ls -t /var/lib/pbench-agent/ | grep "pbench-user" | head -1)"/1
-fi
-
-# copy the prometheus write ahead log from the prometheus pod
 function capture_wal() {
 	echo "================================================================================="
 	echo "               copying prometheus wal from $prometheus_pod                       "
 	echo "================================================================================="
 	oc cp $prometheus_namespace/$prometheus_pod:/prometheus/wal -c prometheus $OUTPUT_DIR/wal/
-	echo "creating a tarball of the captured DB at $OUTPUT_DIR"
 	XZ_OPT=--threads=0 tar cJf $OUTPUT_DIR/prometheus-$ts.tar.xz $OUTPUT_DIR/wal
-	if [[ $? == 0 ]]; then
+	if [[ $? -eq 0 ]]; then
 		rm -rf $OUTPUT_DIR/wal
 	fi
 }
 
-# copy the entire DB from the prometheus pod
+
 function capture_full_db() {
 	echo "================================================================================="
 	echo "            copying the entire prometheus DB from $prometheus_pod                "
 	echo "================================================================================="
 	oc cp  $prometheus_namespace/$prometheus_pod:/prometheus/ -c prometheus $OUTPUT_DIR/data/
-	echo "creating a tarball of the captured DB at $OUTPUT_DIR"
 	XZ_OPT=--threads=0 tar cJf $OUTPUT_DIR/prometheus-$ts.tar.xz -C $OUTPUT_DIR/data .
-	if [[ $? == 0 ]]; then
+	if [[ $? -eq 0 ]]; then
 		rm -rf $OUTPUT_DIR/data
 	fi
 }
 
+
 function must_gather() {
-	echo "================================================================================="
-	echo "                     Running OpenShift must-gather                               "
-	echo "================================================================================="
-	echo "Asssuming the oc client on the host supports collecting must-gather"
 	oc adm must-gather --dest-dir=$OUTPUT_DIR/must-gather-$ts
+	XZ_OPT=--threads=0 tar cJf $OUTPUT_DIR/must-gather-$ts.tar.xz $OUTPUT_DIR/must-gather-$ts
+	if [[ $? -eq 0 ]]; then
+		rm -rf $OUTPUT_DIR/must-gather-$ts
+	fi
+}
+
+
+function post_to_server() {
+	# parameters
+	# 	1 data filepath
+
+	echo "Store on server: $1"
+	curl $STORAGE_MODE/api --form file=@"$1"
+	if [[ $? -eq 0 ]]; then
+		rm -rf "$1"
+	fi
 }
-	
-if [[ "$PROMETHEUS_CAPTURE" == "true" ]]; then
+
+
+function validate_server_is_up() {
+	# return
+	# 	http request code
+
+	curl $STORAGE_MODE 1>/dev/null
```

Comment:
Maybe set the storage_mode to something like data_server and allow user to pass in the address?

Reply:
Changed purpose of STORAGE_MODE added another environment variable to take the data server url, DATA_SERVER_URL